constantine/math_compiler/experimental/backends/cuda.nim

```diff
@@ -224,9 +367,12 @@ proc genCuda*(ctx: var GpuContext, ast: GpuAst, indent = 0): string =
       result = ctx.genCuda(expandedBody, indent)
 
   of gpuBinOp:
-    result = indentStr & "(" & ctx.genCuda(ast.bLeft) & " " &
-             ast.bOp & " " &
-             ctx.genCuda(ast.bRight) & ")"
+    ctx.withoutSemicolon:
```

Comment:
Seems like it was defined in an earlier PR but that's a weird name, does it do string interpolation without a semicolon at the end?

Reply:
When encountering a `gpuBlock` in the code generation, if we are inside of a (nested) `withoutSemicolon` template, we do not emit a `;` at the end of statements in a block.

The idea is that we can have a `gpuBlock` as part of, for example, an if condition. We need a way to suppress the emission of a `;` at the end of each element of the condition.

`withoutSemicolon` simply changes the `skipSemicolon` field of the `GpuContext` until the end of the template. See the codegen part of `gpuBlock` above.